src/QAT/qat/purr/backends/live.py

```diff
@@ -248,46 +248,42 @@ def optimize(self, instructions):
 
             return instructions
 
+    def validate_hardware(self, instructions: List[Instruction]):
```

Comment:
This shouldn't really be a new method unless it's used somewhere I can't see. Whenever you're validating you want to run everything anyway, including the parent method, so I'd like to see it moved back unless there's a strong reason for having a new public one.

Reply:
Reverted